src/Common/src/System/Marvin.cs

```diff
@@ -57,7 +57,7 @@ public static long ComputeHash(ReadOnlySpan<byte> data, ulong seed)
                     break;
 
                 case 2:
-                    p0 += 0x800000u | MemoryMarshal.Cast<byte, ushort>(data)[0];
+                    p0 += 0x800000u | ((uint)data[0]) << 8 | data[1];
```

Comment:
also please use brackets as operator precedence is not obvious

Reply:
Updated in the followup push.